packages/backend-common/src/reading/GithubUrlReader.ts

```diff
@@ -90,16 +90,21 @@ export class GithubUrlReader implements UrlReader {
   }
 
   async readUrl(
-    url: string,
+    originalUrl: string,
     options?: ReadUrlOptions,
   ): Promise<ReadUrlResponse> {
+    const { ref } = parseGitUrl(originalUrl);
+    const wildcardBranchName = ref !== 'qwerty' ? 'qwerty' : 'qwerty1';
+    // ensure wildcardBranchName doesn't clash with real name
+    const url = originalUrl.replace('/-/', `/${wildcardBranchName}/`);
```

Comment:
Feeling it might be nicer to pass on `-` on here instead of the replacement? 😅, then have `getGitHubFileFetchUrl` support that instead

Reply:
it would be nicer. but the need to dummy out '-' is so parseGitUrl works correctly. parseGitUrl is called in getCredentials as well. so we'd have to pass it on to there as well. made more sense to resolve here and solve both problems at once

Reply:
Bit confused, we're calling `parseGitUrl` just above so I assume that works? Or is returning incorrect values for the other pieces?

Tbh I don't think we can go with this workaround, so if there's no other option I think we need to replace `parseGitUrl` with custom parsing logic : /

Reply:
`parseGitUrl ` is called above to check the branch name as we don't want to mistakenly mix up a real branch name for a dummy name.

when there is a wild card there `parseGitUrl ` returns the wrong value. 

i could create a wrapping function that will do this functionality when `parseGitUrl` is called. 

something like:

```typescript

function parseGithubWithWildcard(url){
  const isWildcard = parseGitUrl(url).ref === '';
  if (isWildCard) {
    // replace with dummy and return values;
  }
  return parseGitUrl(url)
 }
```

Reply:
Main thing I want to avoid here is the signature change of `getGitHubFileFetchUrl`. It should be possible to that function to handle `-` without any extra parameters.